bundler/lib/bundler/compact_index_client/parser.rb

```diff
@@ -79,6 +85,17 @@ def parse_version_checksum(line, checksums)
         checksum = line[checksum_start, checksum_end]
         checksums[name.freeze] = checksum # freeze name since it is used as a hash key
       end
+
+      def parse_info_checksum(line, checksums)
+        return unless (name_end = line.index(" ")) # Artifactory bug causes blank lines in artifactor index files
+        checksum_start = name_end + 1
+        checksum_end = line.size - checksum_start
```

Comment:
Here's a chance where we could read until the next space or end of line, instead of just to the end of the line. This would allow extensibility in the future.

Reply:
what extensibility would we need from this file? the other thing I want to test is if making the file be binary saves a significant amount of space, particularly when doing range requests

Reply:
Looks like the savings from going from hexadecimal to binary checksums isnt huge

```
GZIP
	   text_info:    5_020_893
	    versions:    7_205_328
	 binary_info:    4_443_732
raw
	   text_info:    9_273_126
	    versions:   21_012_084
	 binary_info:    5_958_121
```